linera-service/src/cli_wrappers/local_net.rs

```diff
@@ -627,6 +628,34 @@ impl LocalNet {
         self.running_validators.insert(validator, validator_proxy);
         Ok(())
     }
+
+    /// Terminates all the processes of a given `validator`.
+    pub async fn stop_validator(&mut self, validator: usize) -> Result<()> {
+        if let Some(mut validator) = self.running_validators.remove(&validator) {
+            validator.terminate().await?;
+        }
+        Ok(())
+    }
+
+    /// Returns a [`linera_rpc::Client`] to interact directly with a `validator`.
+    pub async fn validator_client(&mut self, validator: usize) -> Result<linera_rpc::Client> {
+        let node_provider = linera_rpc::NodeProvider::new(linera_rpc::NodeOptions {
+            send_timeout: Duration::from_secs(1),
+            recv_timeout: Duration::from_secs(1),
+            retry_delay: Duration::ZERO,
+            max_retries: 0,
+        });
+
+        let port = Self::proxy_port(validator);
+        let schema = match self.network.internal {
+            Network::Grpc | Network::Grpcs => "grpc",
+            Network::Tcp => "tcp",
+            Network::Udp => "udp",
+        };
```

Comment:
That should not be here, but instead in the `cli_wrapper/mod.rs` implementation of `Network`.
We have plenty of such functions, but I think those functions should be in the same place.

Reply:
Good point, thanks!